### PR TITLE
Remove deprecated APIs

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

- Remove `Astro.resolve`, `Astro.fetchContent`, and `Astro.canonicalURL`.
- Remove `buildConfig` option parameter from integration `astro:build:start` hook

The changesets have more information.

## Testing

Existing tests should pass (made some tweaks to the old APIs)

## Docs

Needs a review of the changeset description.

### CODE REVIEW

File: .changeset/two-geese-eat.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+'astro': major
```
Comment (bholmesdev) [inline .changeset/two-geese-eat.md:2]: Hm, somehow got 2 changesets pointing to astro core?

File: packages/astro/src/runtime/server/astro-global.ts [hunk 2]
```diff
@@ -26,28 +19,13 @@ function createAstroGlobFn() {
 // This is used to create the top-level Astro global; the one that you can use
 // Inside of getStaticPaths.
 export function createAstro(
-	filePathname: string,
-	_site: string | undefined,
-	projectRootStr: string
+	_filePathname: string, // legacy unused param from compiler
+	site: string | undefined,
+	_projectRootStr: string // legacy unused param from compiler
```
Comment (bholmesdev) [inline packages/astro/src/runtime/server/astro-global.ts:24]: I assume this is a separate TODO for the compiler?

**Review by bholmesdev — COMMENTED** [review]
A couple questions before merging. Really `depr`appreciate this!

File: .changeset/two-geese-eat.md (hunk 1 above)
Comment (bluwy) [reply .changeset/two-geese-eat.md:2]: I think this is correct as this changeset is a breaking change for the `astro:build:start` hook 🤔  The other breaking change is for Astro globals which I thought would be good to separate.

File: packages/astro/src/runtime/server/astro-global.ts (hunk 2 above)
Comment (bluwy) [reply packages/astro/src/runtime/server/astro-global.ts:24]: Good idea. Let's put a TODO here and make sure we make all the breaking changes for the compiler as we're planning a 1.0 for it too.

File: .changeset/two-geese-eat.md (hunk 1 above)
Comment (bholmesdev) [reply .changeset/two-geese-eat.md:2]: Okay, if that separation resolves then perfect 👍

File: packages/astro/src/runtime/server/astro-global.ts [hunk 3]
```diff
@@ -25,29 +18,15 @@ function createAstroGlobFn() {
...
-	const projectRoot = new URL(projectRootStr);
 	return {
-		site,
+		site: site ? new URL(site) : undefined,
 		generator: `Astro v${ASTRO_VERSION}`,
-		fetchContent: createDeprecatedFetchContentFn(),
 		glob: createAstroGlobFn(),
-		// INVESTIGATE is there a use-case for multi args?
-		// TODO remove in 2.0
-		resolve(...segments: string[]) {
```
Comment (aFuzzyBear) [inline packages/astro/src/runtime/server/astro-global.ts:43]: Ah `Astro.resolve` I mind when you came in, and then boom it was used everywhere, prior to the shift to esm. 
RIP

File: packages/astro/src/runtime/server/astro-global.ts [hunk 4]
```diff
@@ -25,29 +18,15 @@ function createAstroGlobFn() {
...
+	_projectRootStr: string
 ): AstroGlobalPartial {
-	const site = _site ? new URL(_site) : undefined;
-	const referenceURL = new URL(filePathname, `http://localhost`);
-	const projectRoot = new URL(projectRootStr);
 	return {
-		site,
+		site: site ? new URL(site) : undefined,
 		generator: `Astro v${ASTRO_VERSION}`,
-		fetchContent: createDeprecatedFetchContentFn(),
```
Comment (aFuzzyBear) [inline packages/astro/src/runtime/server/astro-global.ts:39]: `Astro.fetchContent` another fan favorite, way before `Astro.glob` got better, this was the trick used on almost every Astro site <0.24.

RIP...

**Review by aFuzzyBear — COMMENTED** [review]
Its a sad PR this one, but needed, thank you for doing this @bluwy great bit of housekeeping 🧹

**Comment by aFuzzyBear** [comment]
Could we look into the smoke test, Im not familiar with this part of the process, but having re-ran them its saying on the docs build an external dep is missing its `exports` 🤷‍♂️

**Comment by bholmesdev** [comment]
@aFuzzyBear we just merged a fix! Needs a rebase and we're golden 🐻

**Review by bholmesdev — APPROVED** [review]

**Review by delucis — APPROVED** [review]
LGTM!

**Pull request merged**